File: client/src/app/models/invoiceHeader.ts
import { InvoiceItem } from "./invoiceItem";
import { User } from "./user";
import { Buyer } from "./buyer";

export class InvoiceHeader {
    id?: number;
    invoiceNumber: number;
    shippingDate: Date;
    documentDate: Date;
    arrivalDate: Date;
    description: string;
    remark: string;
    placeOfIssuance: string;
    dateOfIssuance: Date;
    fiscalNumber: string;
    buyer: Buyer;
    numberOfItems: number;
    totalPrice: number;
    isCharged: boolean;
    appUserId?: number;
    appUser?: User;
    invoiceItems?: InvoiceItem[];
  }
  
  export enum InvoiceStatus {
    Created,
    Charged,
  }

File: API/Entities/InvoiceHeader.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace API.Entities
{
    public class InvoiceHeader
    {
    public int Id { get; set; }
    public int InvoiceNumber { get; set; }
    public DateTime ShippingDate { get; set; }
    public DateTime DocumentDate { get; set; }
    public DateTime ArrivalDate { get; set; }
    public string Description { get; set; }
    public string Remark { get; set; }
    public string PlaceOfIssuance { get; set; }
    public DateTime DateOfIssuance { get; set; }
    public string FiscalNumber { get; set; }
    public Buyer Buyer { get; set; }
    public int NumberOfItems { get; set; }
    public decimal TotalPrice => InvoiceItems.Sum(x => x.TotalPrice);
    public bool IsCharged { get; set; }
    public int AppUserId { get; set; }
    public AppUser AppUser { get; set; }
    public List<InvoiceItem> InvoiceItems { get; set; } = new List<InvoiceItem>();

    }
}